src/eventEmitter/index.js

```diff
@@ -0,0 +1,165 @@
+var
+  uuid = require('uuid');
+
+function EventEmitter(eventTimeout) {
+  Object.defineProperties(this, {
+    eventListeners: {
+      value: {}
+    },
+    eventTimeout: {
+      value: eventTimeout || 200,
+      writeable: false
+    }
+  });
+}
+
+/**
+ * Emit an event to all registered listeners
+ * An event cannot be emitted multiple times before a timeout has been reached.
+ */
+EventEmitter.prototype.emitEvent = function emitEvent(event) {
+  var
+    self = this,
+    now = Date.now(),
+    args = Array.prototype.slice.call(arguments, 1),
+    eventProperties = this.eventListeners[event];
+
+  if (!eventProperties) {
+    return false;
+  }
+
+  if (eventProperties.lastEmitted && eventProperties.lastEmitted >= now - this.eventTimeout) {
```

Comment:
This is one of the specific implementations for Kuzzle but for the main part, the rest of the implementation looks rather standard to me.

[discussion] Would it make sense to rely on an external library (i.e. webpack uses [ev-emitter](https://github.com/metafizzy/ev-emitter/blob/master/ev-emitter.js) by default), and override its functionalities when required?